src/libAtomVM/externalterm.c

```diff
@@ -325,7 +325,11 @@ static term parse_external_terms(const uint8_t *external_term_buf, int *eterm_si
             int32_t value = READ_32_UNALIGNED(external_term_buf + 1);
 
             *eterm_size = 5;
-            return term_from_int32(value);
+            if ((value > 268435455) || (value < -268435455)) {
```

Comment:
I think we should avoid hard coding, I rather go for a constant or something else.

Reply:
I suggest you taking a look to `term_make_maybe_boxed_int64`.

Reply:
Please review fixes.  Thanks!